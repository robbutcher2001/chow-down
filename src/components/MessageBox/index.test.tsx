import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import { NegativeBox, ErrorBox } from '.';

test('NegativeBox basic snapshot render', () => {
  const negativeBox = renderer.create(
    <NegativeBox message='test_negative' />
  );

  expect(negativeBox.toJSON()).toMatchSnapshot();
});

test('NegativeBox correct tag content assertion', () => {
  const { getByText } = render(
    <NegativeBox message='test_negative' />
  );

  expect(getByText(/test_negative/).textContent).toEqual('test_negative');
});

test('ErrorBox basic snapshot render', () => {
  const errorBox = renderer.create(
    <ErrorBox message='test_error' />
  );

  expect(errorBox.toJSON()).toMatchSnapshot();
});

test('ErrorBox correct tag content assertion', () => {
  const { getByText } = render(
    <ErrorBox message='test_error' />
  );

  expect(getByText(/test_error/).textContent).toEqual('test_error');
});