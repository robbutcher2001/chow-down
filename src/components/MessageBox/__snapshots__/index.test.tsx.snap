// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBox basic snapshot render 1`] = `
<section
  className="sc-bdVaJa cIFhlI"
>
  <img
    src="test-file-stub"
  />
  <div
    className="red"
  >
    test_error
  </div>
</section>
`;

exports[`NegativeBox basic snapshot render 1`] = `
<section
  className="sc-bdVaJa cIFhlI"
>
  <img
    src="test-file-stub"
  />
  <div
    className="e"
  >
    test_negative
  </div>
</section>
`;
